cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1103,7 +1330,12 @@ Result<FileInfo> AzureFileSystem::GetFileInfo(const std::string& path) {
 }
 
 Result<FileInfoVector> AzureFileSystem::GetFileInfo(const FileSelector& select) {
-  return Status::NotImplemented("The Azure FileSystem is not fully implemented");
+  Azure::Core::Context context;
+  Azure::Nullable<int32_t> page_size_hint;  // unspecified
+  FileInfoVector results;
+  RETURN_NOT_OK(
+      impl_->GetFileInfoWithSelector(context, page_size_hint, select, &results));
+  return std::move(results);
```

Comment:
This is because I want to move the `results` into the `Result<T>` that is being created.

It's equivalent to this:

```cpp
  Result<FileInfoVector> result{std::move(results)};
  return result;
```

I will change the line to `return {std::move(results)};` to clarify what is going on.

Reply:
I don't think the `std::move` is necessary at all? The moving should happen nevertheless AFAIU.
cc @bkietz for advice.

Reply:
For a counterpoint, when a function returns a `Result<std::unique_ptr<T>>`, you can return a `std::unique_ptr<T>` without moving it explicitly.

Reply:
Right... I keep forgetting that a tail-position `return x` turns `x` into `T&&`.